akka-typed/src/test/java/akka/typed/javadsl/AdapterTest.java

```diff
@@ -0,0 +1,342 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.typed.javadsl;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import scala.concurrent.duration.FiniteDuration;
+import java.util.concurrent.TimeUnit;
+import akka.actor.ActorSystem;
+import akka.testkit.AkkaJUnitActorSystemResource;
+import akka.testkit.AkkaSpec;
+import akka.typed.ActorRef;
+import akka.typed.Behavior;
+import akka.typed.Signal;
+import akka.typed.Terminated;
+import akka.typed.javadsl.Adapter;
+import akka.testkit.javadsl.TestKit;
+import akka.actor.SupervisorStrategy;
+import static akka.typed.javadsl.Actor.*;
+
+public class AdapterTest extends JUnitSuite {
+
+  static akka.actor.Props untyped1() {
+    return akka.actor.Props.create(Untyped1.class, () -> new Untyped1());
+  }
+
+  static class Untyped1 extends akka.actor.AbstractActor {
+    @Override
+    public Receive createReceive() {
+      return receiveBuilder()
+        .matchEquals("ping", s -> getSender().tell("pong", getSelf()))
+        .match(ThrowIt.class, t -> {
+          throw t;
+        })
+        .build();
+    }
+  }
+
+  static class Typed1 {
+    private final akka.actor.ActorRef ref;
+    private final akka.actor.ActorRef probe;
+
+    private Typed1(akka.actor.ActorRef ref, akka.actor.ActorRef probe) {
+      this.ref = ref;
+      this.probe = probe;
+    }
+
+    static Behavior<String> create(akka.actor.ActorRef ref, akka.actor.ActorRef probe) {
+      Typed1 logic = new Typed1(ref, probe);
```

Comment:
The reason I created this `Typed1` class was to be able to capture the `ref` and `probe` parameters, to be used from the `onMessage` and `onSignal` methods. I didn't want to write everything in two big lambdas. Well, now there is two big methods, but you get the point.